signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
> isn't this a separate issue though? I agree we need to verify the user provided digest against what we retrieve from the network though.

Yes, it is orthogonal to this PR.

> Question, how do we handle this for matchAnyTag?

Well, the signature doesn’t just contain a tag which we happily apply to any manifest; it also contains a manifest digest. We will accept a signature which says `sha256:HEX_1` and `testimage:anythingwedontcare`. The “anythingwedontcare” is the “any tag” in this terminology.

> What happen otherwise?

What is the “otherwise” case?

> Also, why do we call it UntaggedReference? an user can't initiate a pull w/o a valid tag or reference.

Well, it is about a reference, and the reference in this case doesn’t have a tag. (I guess `atomic pull` doesn’t work right now if there isn’t a tag, if that’s what you mean, but things like `skopeo copy docker:// docker://` generally do.) I am definitely open to other suggestions.

> We should call it CanonicalReference maybe.

Canonical is jargon, and more importantly it is not necessarily obvious to everyone that it means “with a `@digest". `signedIdentityWhenReferencingByDigest`…no.